### ISSUE TITLE
option to auto-compress on sync

### ISSUE THREAD

**Issue by kevgo** [git-town/git-town#3320]
_Extracted from @tranhl in https://github.com/git-town/git-town/issues/2086#issuecomment-2051230333_
            
> Would it be possible to have an option to auto-compress commits when syncing stacks? That would help nicely with #3236 and the new rebase sync strategy behaviours introduced in v13!

**Comment by kevgo** [comment]
Possible names for the config option: 
- `auto-compress`
- `compress-on-sync`
- `sync-compress`

How this would look in the Git configuration:

```
git-town.auto-compress=true
git-town.compress-on-sync=true
git-town.sync-compress=true
```

In the config file

```toml
compress-on-sync = true
push-hook = true
push-new-branches = true
ship-delete-tracking-branch = false
sync-compress = true
sync-tags = false
sync-upstream = true
```

**Comment by kevgo** [comment]
`auto-compress` sounds like the best option here.

**Comment by kevgo** [comment]
Rather than syncing first using a particular sync-strategy, and then squashing (which makes the sync-strategy used irrelevant), why not simply do this on a branch to sync-and-squash:

```
git reset <parent>
```

This (hopefully) updates the branch and squashes the commits at the same time.

If this works, there could be a new sync-strategy variant called `squash`, rather than an `auto-compress` option.

**Comment by kevgo** [comment]
A better name for the new sync strategy might be "compress".

- it matches the `git town compress` command, and pretty much just applies it after syncing
- "squash" has a specific meaning in Git. In interactive rebases it means "fold this commit into another commit but keep its commit message". When auto-squashing it means the same thing but the trigger to squash comes from the commit message.
- A more correct name would be "fixup" but that also implies that a specific commit gets folded into another existing commit. The new sync strategy does something different: it folds all commits in the branch except the first one into the first commit of the branch, keeping only the commit message of the first commit

I therefore think a new term for the sync strategy ("compress") is justified.

**Comment by kevgo** [comment]
This is now released in [v15.2](https://github.com/git-town/git-town/releases/tag/v15.2.0). Check it out 😃
